app/routes/yara_rules.py

```diff
@@ -562,3 +567,101 @@ def delete_all_inactive_yara_rules():
     db.session.query(yara_rule.Yara_rule).filter(yara_rule.Yara_rule.active == 0).delete()
     db.session.commit()
     return jsonify(''), 200
+
+
+@app.route('/ThreatKB/yara_rules/<int:yara_rule_id>/revert-to-revision/<int:revision>', methods=['PUT'])
+@auto.doc()
+@login_required
+def revert_yara_rule_to_revision(yara_rule_id, revision):
+    """Revert given yara rule to provided revision number
+    Return: yara_rule string representation"""
+
+    current_entity = yara_rule.Yara_rule.query.get(yara_rule_id)
+    revision_entity = Yara_rule_history.query \
+        .filter_by(yara_rule_id=yara_rule_id) \
+        .filter_by(revision=revision) \
+        .first()
+
+    if not current_entity or not revision_entity:
+        abort(404)
+
+    if not current_user.admin and current_entity.owner_user_id != current_user.id:
+        abort(403)
+
+    revision_dict = revision_entity.to_dict()
+    yara_revision_dict = revision_dict["rule_json"]
+
+    temp_sig_id = current_entity.eventid
+    if not current_entity.category == yara_revision_dict['category']:
+        temp_sig_id = CfgCategoryRangeMapping.get_next_category_eventid(yara_revision_dict['category'])
```

Comment:
Let me know if this event ID logic is correct. But essentially, whenever the category doesn't match between current and the previous revision you wish to revert to, a new category event ID is retrieved and used.